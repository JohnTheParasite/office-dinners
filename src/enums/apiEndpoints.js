export class ApiEndpoints {
  static LOGIN = process.env.VUE_APP_API_URL + "/login"
  static USER_LIST = process.env.VUE_APP_API_URL + "/user/list"
  static CREATE_USER = process.env.VUE_APP_API_URL + "/user/create"
  static USER_DATA = process.env.VUE_APP_API_URL + "/user/update"
  static CAFE_LIST = process.env.VUE_APP_API_URL + "/cafe/list"
  static TOGGLE_CAFE = process.env.VUE_APP_API_URL + "/cafe/toggle"
  static CREATE_CAFE = process.env.VUE_APP_API_URL + "/cafe/create"
  static CAFE_DATA = process.env.VUE_APP_API_URL + "/cafe/update"
  static CAFE_COMMENTS = process.env.VUE_APP_API_URL + "/comment/cafe"
  static CREATE_COMMENT = process.env.VUE_APP_API_URL + "/comment/add"
  static VOTE = process.env.VUE_APP_API_URL + "/cafe/vote"
  static CAFE_HEADER = process.env.VUE_APP_API_URL + "/cafe/header-data"
  static VOTES_OPEN = process.env.VUE_APP_API_URL + "/cafe/open-votes"
  static VOTES_CLOSE = process.env.VUE_APP_API_URL + "/cafe/close-votes"
  static VOTES_AUTOCLOSE = process.env.VUE_APP_API_URL + "/cafe/set-auto-close-time"
  static ORDER_ADD = process.env.VUE_APP_API_URL + "/cafe/add-order"
  static ORDER_CHANGE = process.env.VUE_APP_API_URL + "/order/change-order"
  static ORDERS_GET = process.env.VUE_APP_API_URL + "/order/get-for-date"
  static SET_ORDER_USER_TO_PAY = process.env.VUE_APP_API_URL + "/order/select-user-to-pay"
  static CLOSE_ORDER = process.env.VUE_APP_API_URL + "/order/close-order"
  static OPEN_ORDER = process.env.VUE_APP_API_URL + "/order/open-order"
  static SET_ORDER_ORDERED = process.env.VUE_APP_API_URL + "/order/set-ordered"
  static DELETE_ORDER = process.env.VUE_APP_API_URL + "/order/delete"
  static UPDATE_ORDER_PRICES = process.env.VUE_APP_API_URL + "/order/update-order-prices"
  static SET_ORDER_DISCOUNT = process.env.VUE_APP_API_URL + "/order/set-order-discount"
}
