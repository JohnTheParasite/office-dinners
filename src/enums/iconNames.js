export default class IconNames {
  static EYE = "fa-eye"
  static EYE_SLASH = "fa-eye-slash"
}
