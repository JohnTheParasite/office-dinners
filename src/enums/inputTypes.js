export default class InputTypes {
  static TEXT = "text"
  static PASSWORD = "password"
}
