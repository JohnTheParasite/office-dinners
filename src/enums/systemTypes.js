export default class SystemTypes {
  static DEFAULT = "default"
  static SUCCESS = "success"
  static DANGER = "danger"
  static WARNING = "warning"
  static PRIMARY = "primary"
}
