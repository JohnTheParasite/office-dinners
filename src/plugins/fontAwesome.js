import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
