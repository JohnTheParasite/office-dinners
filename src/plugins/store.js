import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
