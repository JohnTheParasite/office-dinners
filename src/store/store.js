import Vue from "vue"
import Vuex from "vuex"
import global from "./modules/global"
import toasts from "./modules/toasts"

Vue.use(Vuex)

export default new Vuex.Store({
  strict: true,
  state: {},
  mutations: {},
  actions: {},
  modules: {
    global,
    toasts
  }
})
