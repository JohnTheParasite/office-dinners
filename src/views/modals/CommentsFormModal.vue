<template>
  <b-modal id="commentsDataModal" :header-class="variation" hide-footer centered>
    <template #modal-header="{ close }">
      <h5>{{ $t("cafe.comments") }}</h5>
      <form-button @click="close" type="secondary" class="close">
        <fa-icon icon="close" />
      </form-button>
    </template>
    <div class="new-comment">
      <text-input label="user.name" :init-value="formGroup.username" @input="onChange('username', $event)" :required="true" />
      <div class="comment-grade">
        <b>{{ $t("cafe.food") }}</b>
        <b-form-rating v-model="formGroup.rating_food"></b-form-rating>
      </div>
      <div class="comment-grade">
        <b>{{ $t("cafe.delivery") }}</b>
        <b-form-rating v-model="formGroup.rating_delivery"></b-form-rating>
      </div>
      <text-input label="Comment" :multiline="true" class="comment" :init-value="formGroup.comment" @input="onChange('comment', $event)" />
      <form-button :disabled="!verified" label="cafe.addComment" @click="addComment" />
      <div class="separator"></div>
    </div>
    <div class="comments">
      <div class="comment-container" v-for="comment in comments" :key="comment.id">
        <div class="commentator-name">{{ comment.username }}</div>
        <div class="comment-date">{{ comment.create_date }}</div>
        <div class="comment-grade">
          <b>{{ $t("cafe.food") }}</b>
          <b-form-rating v-model="comment.rating_food" readonly></b-form-rating>
        </div>
        <div class="comment-grade">
          <b>{{ $t("cafe.delivery") }}</b>
          <b-form-rating v-model="comment.rating_delivery" readonly></b-form-rating>
        </div>
        <div class="comment">{{ comment.comment }}</div>
      </div>
    </div>
  </b-modal>
</template>

<script>
import FormButton from "@/components/controls/FormButton"
import SystemTypes from "@/enums/systemTypes"
import FaIcon from "@/components/icons/FaIcon"
import ApiErrorHelper from "@/services/apiErrorHelper"
import TextInput from "@/components/controls/TextInput"
import { ApiEndpoints } from "@/enums/apiEndpoints"
import FormDataService from "@/services/formDataService"

export default {
  name: "CommentsFormModal",
  components: { TextInput, FaIcon, FormButton },
  mixins: [ApiErrorHelper],
  props: {
    variation: {
      default: SystemTypes.PRIMARY
    }
  },
  data() {
    return {
      formGroup: {
        username: "Anonymous",
        cafe_id: "",
        rating_food: 0,
        rating_delivery: 0,
        comment: ""
      },
      comments: [],
      cafeId: undefined
    }
  },
  methods: {
    show(cafeId) {
      this.cafeId = cafeId
      this.initFormGroup()
      this.getCafeComments(cafeId)
      this.$bvModal.show("commentsDataModal")
    },
    onChange(field, value) {
      this.formGroup[field] = value
    },
    initFormGroup() {
      this.formGroup = {
        username: "Anonymous",
        cafe_id: this.cafeId,
        rating_food: 0,
        rating_delivery: 0,
        comment: ""
      }
      this.comments = []
    },
    addComment() {
      this.$axios
        .post(ApiEndpoints.CREATE_COMMENT, FormDataService.getFormData(this.formGroup))
        .then((response) => {
          this.$store.commit("toasts/addSuccessToast", "cafe.commentAdded")
          this.$emit("refreshTable")
          this.comments = response.data
          this.initFormGroup()
        })
        .catch((error) => {
          this.catchAxiosError(error)
        })
    },
    getCafeComments(cafeId) {
      this.$axios
        .get(ApiEndpoints.CAFE_COMMENTS + "/" + cafeId)
        .then((response) => {
          if (response && response.data) {
            this.comments = response.data
          }
        })
        .catch((error) => {
          this.catchAxiosError(error)
        })
    }
  },
  computed: {
    verified() {
      return this.formGroup.username.length > 0 && this.formGroup.rating_food > 0 && this.formGroup.rating_delivery > 0
    }
  }
}
</script>

<style lang="scss" scoped>
@import "../../scss/components/color";

.modal-body {
  .comment-grade {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .comment {
    margin-top: 1rem;
  }

  .comments {
    display: flex;
    flex-direction: column;
    gap: 1rem;

    .comment-container {
      border-radius: 0.428rem;
      box-shadow: $box-shadow;
      padding: 1rem;
      background-color: $light-grey;
      border: 1px solid $primary;

      .commentator-name {
        font-size: 1.2rem;
        font-weight: 500;
      }

      .comment-date {
        font-size: 0.85rem;
        margin-bottom: 1rem;
      }
    }
  }

  .btn {
    width: 100%;
  }

  .separator {
    margin-top: 1rem;
    margin-bottom: 1rem;
  }
}
</style>
