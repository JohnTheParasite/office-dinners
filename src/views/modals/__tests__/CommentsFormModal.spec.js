import CommentsFormModal from "@/views/modals/CommentsFormModal"
import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import i18n from "@/i18n"
import { BootstrapVue } from "bootstrap-vue"
import store from "@/store/store"
import "@/plugins/fontAwesome"
import axios from "axios"
import { ApiEndpoints } from "@/enums/apiEndpoints"

describe("CommentsFormModal", () => {
  let localVue = new createLocalVue()
  localVue.use(i18n)
  localVue.use(BootstrapVue)

  const axios = require("axios")
  jest.mock("axios")

  it("should render CommentsFormModal and compute filled fields", () => {
    let commentsModal = mount(CommentsFormModal, { localVue, i18n, store })
    expect(commentsModal).toMatchSnapshot()

    expect(commentsModal.vm.verified).toBeFalsy()

    commentsModal.vm.formGroup.username = "Anonymous"
    commentsModal.vm.formGroup.rating_food = 5
    commentsModal.vm.formGroup.rating_delivery = 5

    expect(commentsModal.vm.verified).toBeTruthy()
  })

  it("should test onChange() and resetFormGroup() method", () => {
    let commentsModal = shallowMount(CommentsFormModal, { localVue, i18n, store })

    expect(commentsModal.vm.formGroup.username).toBe("Anonymous")
    commentsModal.vm.onChange("username", "John")
    expect(commentsModal.vm.formGroup.username).toBe("John")
    commentsModal.vm.resetFormGroup()
    expect(commentsModal.vm.formGroup.username).toBe("Anonymous")
  })

  it("should test show() and getCafeComments() method", async () => {
    let commentsModal = mount(CommentsFormModal, { localVue, i18n, store, mocks: { $axios: axios } })
    expect(commentsModal.vm.cafeId).toBe(undefined)
    axios.get.mockResolvedValueOnce({ data: [] })
    await commentsModal.vm.show("15")
    expect(axios.get).toHaveBeenCalledWith(ApiEndpoints.CAFE_COMMENTS + "/15")
    expect(commentsModal.vm.cafeId).toBe("15")
    expect(commentsModal.vm.$children[0].isVisible).toBeTruthy()
    axios.get.mockRejectedValueOnce(false)
    commentsModal.vm.getCafeComments("15")
    expect(axios.get).toHaveBeenCalledWith(ApiEndpoints.CAFE_COMMENTS + "/15")
  })

  it("should test addComment() method", async () => {
    let commentsModal = shallowMount(CommentsFormModal, { localVue, i18n, store, mocks: { $axios: axios } })

    axios.post.mockResolvedValueOnce({ data: [] })
    await commentsModal.vm.addComment()
    expect(axios.post).toBeCalledWith(ApiEndpoints.CREATE_COMMENT, expect.anything())

    axios.post.mockRejectedValueOnce(false)
    await commentsModal.vm.addComment()
    expect(axios.post).toBeCalledWith(ApiEndpoints.CREATE_COMMENT, expect.anything())
  })
})
