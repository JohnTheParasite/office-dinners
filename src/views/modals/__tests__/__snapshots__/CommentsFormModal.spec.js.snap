// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsFormModal should render CommentsFormModal and compute filled fields 1`] = ``;
